velox/CMakeLists.txt

```diff
@@ -32,6 +32,6 @@ add_subdirectory(vector)
 add_subdirectory(row)
 add_subdirectory(flag_definitions)
 add_subdirectory(codegen)
-if(${CODEGEN_SUPPORT})
+if(${CODEGEN_SUPPORT} AND ${VELOX_BUILD_TESTING})
```

Comment:
codegen code is experimental, but it isn't testing code. One flag shouldn't impact other in this case, IMHO.

Reply:
codegen has a target called velox_all_link which has some deps on velox_exec_test_util. So it will fail if don't build. So I add the condition here.